src/main/commandServer/__tests__/settingsValidator.spec.ts

```diff
@@ -123,50 +127,56 @@ describe(SettingsValidator, () => {
       ];
 
       for (const [specifiedSettingSegment, fullQualifiedPreferenceName] of valuesToChange) {
-        const [needToUpdate, errors] = subject.validateSettings(cfg, _.merge({}, cfg, specifiedSettingSegment));
+        const { needToUpdate, needToRestart, errors } = subject.validateSettings(cfg, _.merge({}, cfg, specifiedSettingSegment));
 
         expect(needToUpdate).toBeFalsy();
+        expect(needToRestart).toBeFalsy();
         expect(errors).toHaveLength(1);
         expect(errors).toEqual([`Changing field ${ fullQualifiedPreferenceName } via the API isn't supported.`]);
       }
     });
 
     it('should complain about invalid fields', () => {
-      const [needToUpdate, errors] = subject.validateSettings(cfg, {
+      const { needToUpdate, needToRestart, errors } = subject.validateSettings(cfg, {
         kubernetes: {
           version:         '1.1.1',
           containerEngine: '1.1.2',
           enabled:         1,
           options:         { flannel: 1 },
-        }
+        },
+        noModalDialogs: 'not a boolean',
       });
 
       expect(needToUpdate).toBeFalsy();
+      expect(needToRestart).toBeFalsy();
       expect(errors).toEqual([
         'Kubernetes version "1.1.1" not found.',
         "Invalid value for kubernetes.containerEngine: <1.1.2>; must be 'containerd', 'docker', or 'moby'",
         'Invalid value for kubernetes.enabled: <1>',
         'Invalid value for kubernetes.options.flannel: <1>',
+        'Invalid value for noModalDialogs: <not a boolean>',
       ]);
     });
 
     it('complains about mismatches between objects and scalars', () => {
-      let [needToUpdate, errors] = subject.validateSettings(cfg, { kubernetes: 5 });
+      let result = subject.validateSettings(cfg, { kubernetes: 5 });
 
-      expect(needToUpdate).toBeFalsy();
-      expect(errors).toHaveLength(1);
-      expect(errors[0]).toContain('Setting kubernetes should wrap an inner object, but got <5>');
+      expect(result.needToUpdate).toBeFalsy();
+      expect(result.needToRestart).toBeFalsy();
+      expect(result.errors).toHaveLength(1);
+      expect(result.errors[0]).toContain('Setting kubernetes should wrap an inner object, but got <5>');
 
-      [needToUpdate, errors] = subject.validateSettings(cfg, {
+      result = subject.validateSettings(cfg, {
```

Comment:
FYI, this can be destructured as well, if we do:
```ts
({ needToUpdate, needToRestart, errors } = subject.validateSettings(…));

Reply:
So that's how it's done...

Reply:
But it's reverted for now.